conda/recipes/libkvikio/recipe.yaml

```diff
@@ -0,0 +1,183 @@
+        set -x
+        export CFLAGS=$(echo $CFLAGS | sed -E 's@\-fdebug\-prefix\-map[^ ]*@@g')
+        export CXXFLAGS=$(echo $CXXFLAGS | sed -E 's@\-fdebug\-prefix\-map[^ ]*@@g')
+        set +x
+
+        ./build.sh -v -n libkvikio
+      secrets:
+        - AWS_ACCESS_KEY_ID
+        - AWS_SECRET_ACCESS_KEY
+        - AWS_SESSION_TOKEN
+      env:
+        CMAKE_C_COMPILER_LAUNCHER: ${{ env.get("CMAKE_C_COMPILER_LAUNCHER") }}
+        CMAKE_CUDA_COMPILER_LAUNCHER: ${{ env.get("CMAKE_CUDA_COMPILER_LAUNCHER") }}
+        CMAKE_CXX_COMPILER_LAUNCHER: ${{ env.get("CMAKE_CXX_COMPILER_LAUNCHER") }}
+        CMAKE_GENERATOR: ${{ env.get("CMAKE_GENERATOR") }}
+        PARALLEL_LEVEL: ${{ env.get("PARALLEL_LEVEL") }}
+        RAPIDS_ARTIFACTS_DIR: ${{ env.get("RAPIDS_ARTIFACTS_DIR") }}
+        SCCACHE_BUCKET: ${{ env.get("SCCACHE_BUCKET") }}
+        SCCACHE_IDLE_TIMEOUT: ${{ env.get("SCCACHE_IDLE_TIMEOUT") }}
+        SCCACHE_REGION: ${{ env.get("SCCACHE_REGION") }}
+        SCCACHE_S3_USE_SSL: ${{ env.get("SCCACHE_S3_USE_SSL") }}
+        SCCACHE_S3_NO_CREDENTIALS: ${{ env.get("SCCACHE_S3_NO_CREDENTIALS") }}
+        SCCACHE_S3_KEY_PREFIX: libkvikio/${{ env.get("RAPIDS_CONDA_ARCH") }}/cuda${{ cuda_major }}
+  requirements:
+    build:
+      - ${{ compiler("c") }}
+      - ${{ compiler("cxx") }}
+      - ${{ compiler("cuda") }}
+      - cuda-version =${{ cuda_version }}
+      - cmake ${{ cmake_version }}
+      - ninja
+      - ${{ stdlib("c") }}
+    host:
+      - cuda-version =${{ cuda_version }}
+      - libcurl ${{ libcurl_version }}
+      - if: cuda_major == "11"
+        then:
+          - cudatoolkit
+          - if: linux64
+            then:
+              - libcufile =${{ cuda11_libcufile_host_version }}
+              - libcufile-dev =${{ cuda11_libcufile_host_version }}
+        else: libcufile-dev
+
+outputs:
+  - package:
+      name: libkvikio
+      version: ${{ version }}
+    build:
+      script:
+        content: |
+          cmake --install cpp/build
+      string: cuda${{ cuda_major }}_${{ date_string }}_${{ head_rev }}
+      dynamic_linking:
+        overlinking_behavior: "error"
+      prefix_detection:
+        ignore:
+          - lib/libkvikio.so
+    requirements:
+      build:
+        - cmake ${{ cmake_version }}
+        - ${{ compiler("c") }}
+      host:
+        - cuda-version =${{ cuda_version }}
+        - libcurl ${{ libcurl_version }}
+      run:
+        - ${{ pin_compatible("cuda-version", upper_bound="x", lower_bound="x") }}
+        - if: cuda_major == "11"
+          then:
+            - cudatoolkit
+            - if: linux64
+              then:
+                - libcufile ${{ cuda11_libcufile_run_version }}
+                - libcufile-dev ${{ cuda11_libcufile_run_version }}
+          else:
+            - if: arm64
+              then:
+                - if: match(cuda_version, ">12.1")
+                  then: libcufile
+                  else: libcufile-dev
+              else: libcufile
```

Comment:
Can you help me understand what this is supposed to do? This will give `libcufile` for all CUDA 12 and linux64 jobs, and CUDA 11 ARM jobs will get `libcufile-dev`. But how does that work? There is no libcufile on ARM for CUDA 11 at all.

Reply:
The first builds of `libcufile` for ARM came with CUDA 12.2.

If we could break `libkvikio` into more build variants, this is what would be correct:
- CUDA 11 linux64 (has libcufile-dev at build time, libcufile at runtime)
- CUDA 12 linux64 (has libcufile-dev at build time, libcufile at runtime)
- CUDA 11 arm64 (no libcufile at build time or runtime)
- CUDA 12.0 / 12.1 arm64 (no libcufile at build time or runtime)
- CUDA 12.2+ arm64 (has libcufile-dev at build time, libcufile at runtime)

We've been adding libcufile to all CUDA 12 arm64 builds, so effectively "CUDA 12.0 / 12.1 arm64" has been fetching libcufile from CUDA 12.4 off the `nvidia` channel at runtime, and "CUDA 12.2+ arm64" has been finding the libcufile with a matching `cuda-version` on `conda-forge`.

Reply:
If we retain the status quo, where we have a `libcufile` dependency that is being satisfied by the `nvidia` channel, we won't be able to drop the `nvidia` channel in the future when we drop CUDA 11. It's fundamentally incorrect anyway, the only reason that it finds `libcufile` from the `nvidia` channel is because (a) that channel is in the list of channels and (b) the `nvidia` channel packages aren't correctly constrained by `cuda-version` until CUDA 12.5. That's why it gets the `libcufile` build on the `nvidia` channel that shipped with CUDA 12.4.

Reply:
We don't want to drop libcufile support on CUDA 12 ARM entirely, which is why we've accepted the status quo thus far. However, we may need to invest in making a second build variant of the CUDA 12 ARM package -- one for CUDA 12.0/12.1 and one for CUDA 12.2+. That's the only way for us to package this "correctly" while still keeping support for libcufile on CUDA 12 ARM platforms.

Reply:
The other important consideration is that the `match` syntax doesn't really help us that much because it is evaluated at build time -- we are only doing builds with CUDA 12.8 right now. It might help us if we added a second build that uses CUDA 12.0 or 12.1, but we would need for the `else:` to be empty (no `libcufile` packages on CUDA 12.0/12.1, whereas it would currently use `libcufile-dev`).

Reply:
I know this is a huge wall of text, I just needed to write it out for my own sanity (it's hard to communicate the complexity of this problem/solution space). If you want to defer on this until later, I am okay with keeping the status quo where some CUDA 12.0/12.1 ARM configurations end up pulling a library from the `nvidia` channel. It's not desirable, and it won't work with strict priority, but it would let us unblock rattler-build work.

Reply:
Added a follow-up item to rapidsai/build-planning#47 with a link to this discussion.